applications/minotari_merge_mining_proxy/src/proxy/mod.rs

```diff
@@ -0,0 +1,265 @@
+    use reqwest::Client;
+    use serde_json::json;
+
+    use crate::proxy::{monerod_method::MonerodMethod, utils::convert_reqwest_response_to_hyper_json_response};
+
+    async fn get_json_rpc(
+        method: MonerodMethod,
+        json_rpc_port: u16,
+        hyper_json_response: bool,
+    ) -> Result<String, Error> {
+        match method {
+            MonerodMethod::GetHeight => get_response(json_rpc_port, method).await,
+            MonerodMethod::GetBlockTemplate |
+            MonerodMethod::GetVersion |
+            MonerodMethod::SubmitBlock |
+            MonerodMethod::GetBlockHeaderByHash |
+            MonerodMethod::GetLastBlockHeader => json_rpc_request(method, json_rpc_port, hyper_json_response).await,
+            _ => Err(anyhow!("'{}' not supported", method)),
+        }
+    }
+
+    async fn get_response(json_rpc_port: u16, method: MonerodMethod) -> Result<String, Error> {
+        let full_address = format!("http://127.0.0.1:{}", json_rpc_port);
+        match reqwest::get(format!("{}/{}", full_address, method))
+            .await
+            .unwrap()
+            .json::<serde_json::Value>()
+            .await
+        {
+            Ok(val) => {
+                println!("{}: {}", method, val);
+                Ok(val.to_string())
+            },
+            Err(e) => Err(e.into()),
+        }
+    }
+
+    async fn json_rpc_request(
+        method: MonerodMethod,
+        json_rpc_port: u16,
+        hyper_json_response: bool,
+    ) -> Result<String, Error> {
+        let rpc_method = format!("{}", method);
+        let request_body = match method {
+            MonerodMethod::GetBlockTemplate => json!({
+                "jsonrpc": "2.0",
+                "id": "0",
+                "method": rpc_method,
+                "params": {
+                    "wallet_address": "489r43gR8bDMJNBf4Q6sL9CNERvZQrTqjRCSESqgWQEWWq2UGAfj2voaw3zBtD7U8CQ391Nc1PDHUHiN85yhbZnCDasqzyX",
+                }
+            }),
+            MonerodMethod::GetVersion => json!({
+                "jsonrpc": "2.0",
+                "id": "0",
+                "method": rpc_method,
+                "params": {}
+            }),
+            MonerodMethod::SubmitBlock => json!({
+                "jsonrpc": "2.0",
+                "id": "0",
+                "method": rpc_method,
+                "params": ["1010c9a5cdbc062c8c6fc9c6b0d01299a8b4dd023dc4ee28d7f890deeae012ec11b04f0e3a24c4dc1e180002a4b1cb0101ffe8b0cb0101e0dde1dabd110323fd3161ea3cca2ad8208e66705d4f0ca9de911e99300ac813142891f000ccd1074e0321008fbbaf1ee6c5337eb65241e3f079ddf141c9a916f4b9bf3270b19ea4902c976b01cdf43802b1e9465e4bb2a6c9fc6e281d7b4f2b1aa819acfd8e6133446c895a550208b2f23f2262ef83aa000483b08f79f3bcc0a58d4c303d67db5839505eed4ed7273f6e5fc331c991011772ac88c18fd7f1c9f19b041b128fe923d95c2aafb1dd106bc781a61a2bfa1c66cb8e6dab82d22909b40bda27ec0e96aa0c6c0012023d353f3be941eb6ef1793cad0c773379302ec1567eb4d688b4e66499eee1855ffd97b966eb929ed39c1d056c"]
```

Comment:
How is this derived? The code needs to be present and verifiable

Reply:
what is in here?

Reply:
ok cool so this is just for testing responses, and is a captured response